src/main/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -161,6 +167,55 @@ protected ProcurementCardDocument createProcurementCardDocument(List transaction
         return pcardDocument;
     }
     
+    /**
+     * 
+     * @see org.kuali.kfs.fp.batch.service.impl.ProcurementCardCreateDocumentServiceImpl#createTransactionDetailRecord(org.kuali.kfs.fp.document.ProcurementCardDocument, org.kuali.kfs.fp.businessobject.ProcurementCardTransaction, java.lang.Integer)
+     */
+    @Override
+    protected String createTransactionDetailRecord(ProcurementCardDocument pcardDocument, ProcurementCardTransaction transaction, Integer transactionLineNumber) {
+      String errorText = super.createTransactionDetailRecord(pcardDocument, transaction, transactionLineNumber);
+      
+      // create the Extension object, which contains the Level 3 info
+      createProcurementCardTransactionDetailExtension(transaction, 
+                                                      (ProcurementCardTransactionDetail) pcardDocument.getTransactionEntries()
+                                                                                                      .get(pcardDocument.getTransactionEntries().size() - 1));
+      
+      return errorText;
+    }
+
+
+    protected void createProcurementCardTransactionDetailExtension(ProcurementCardTransaction transaction, ProcurementCardTransactionDetail transactionDetail) {
+      ProcurementCardTransactionDetailExtendedAttribute detailExtension;
+      if (ObjectUtils.isNull(transactionDetail.getExtension())) {
+          detailExtension = new ProcurementCardTransactionDetailExtendedAttribute();
+      } else {
+          detailExtension = (ProcurementCardTransactionDetailExtendedAttribute) transactionDetail.getExtension();
+      }
+
+      if (ObjectUtils.isNotNull(transaction.getExtension())) {

```

Comment:
yeah, sorry, that was me reading too quickly and thinking the objects being checked for null/not null where the same when they're not; my bad